sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/test/TestTableProvider.java

```diff
@@ -150,12 +153,30 @@ public BeamTableStatistics getTableStatistics(PipelineOptions options) {
       return begin.apply(Create.of(tableWithRows.rows).withCoder(rowCoder()));
     }
 
+    @Override
+    public PCollection<Row> buildIOReader(
+        PBegin begin, BeamSqlTableFilter filters, List<String> fieldNames) {
```

Comment:
It would be good to add an assert that `filters instanceof DefaultTableFilter` here as well.

Reply:
Added a check for when to return a result of a regular `buildIOReader`.